libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance.hs

```diff
@@ -426,3 +503,106 @@ instance
       <$> toAgdaTxBody @era tx
       <*> toSpecRep (wits tx)
       <*> toSpecRep (auxiliaryData tx)
+
+trySubmitTxConform_ :: Tx Conway -> ImpTestM Conway ()
+trySubmitTxConform_ = void . trySubmitTxConform
+
+trySubmitTxConform ::
+  Tx Conway ->
+  ImpTestM
+    Conway
+    ( Either
+        [PredicateFailure (EraRule "LEDGER" Conway)]
+        (TxId (EraCrypto Conway))
+    )
+trySubmitTxConform txPreFixup = do
+  nes <- getsNES id
+  doFixup <- gets $ view impDoFixupL
+  tx <-
+    if doFixup
+      then fixupFees txPreFixup
+      else pure txPreFixup
+  let
+  agdaUtxoState <- expectRight . toSpecRep $ nes ^. nesEsL . esLStateL . lsUTxOStateL
+  agdaTx <- expectRight $ toSpecRep tx
+  lastTick <- gets $ view impLastTickL
+  pParams <- getsNES $ nesEsL . curPParamsEpochStateL
+  certState <- getsNES $ nesEsL . esLStateL . lsCertStateL
+  let
+    utxoEnv =
+      UtxoEnv
+        { ueSlot = lastTick
+        , uePParams = pParams
+        , ueGenDelegs = certState ^. certDStateL . dsGenDelegsL
+        , ueCertState = certState
+        }
+  agdaUtxoEnv <- expectRight $ toSpecRep utxoEnv
+  let
+    newAgdaUtxoState = Agda.utxoStep agdaUtxoEnv agdaUtxoState agdaTx
+  -- We already applied the fixup above, so no point doing it again
+  submitRes <- withNoFixup $ trySubmitTx tx
```

Comment:
I still don't understand, why this would not work? Why do we need the whole mutable variable in the state for the fixup?
```suggestion
  submitRes <- trySubmitTx False tx
```